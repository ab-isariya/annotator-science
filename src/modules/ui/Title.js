import PropTypes from 'prop-types';

const Title = ({children, className, size}) => {
  const baseFonts = 'font-inter font-light';

  if (size === 'xl') {
    return (
      <h1 className={`${baseFonts} text-2xl leading-9 ${className}`}>
        {children}
      </h1>
    );
  } else if (size === 'lg') {
    return (
      <h2 className={`${baseFonts} text-base leading-6 ${className}`}>
        {children}
      </h2>
    );
  }

  return (
    <p className={`${baseFonts} text-base leading-6 ${className}`}>
      {children}
    </p>
  );
};

Title.defaultProps = {
  size: 'xl'
};

Title.propTypes = {
  /** Additional classes applied to title */
  className: PropTypes.string,
  /** React node children */
  children: PropTypes.node,
  /** Size: xl=h1, lg=h2, md=h3, sm=h4, xs=h5 */
  size: PropTypes.oneOf(['xl', 'lg', 'md', 'sm', 'xs'])
};

export default Title;
