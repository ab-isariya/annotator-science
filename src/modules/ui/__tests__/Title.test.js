import Title from '../Title';

import renderer from 'react-test-renderer';

describe('<Title/>', () => {
  it('Renders without crashing', () => {
    const comp = renderer.create(<Title />).toJSON();

    expect(comp).toMatchSnapshot();
  });

  it('Renders with size xl', () => {
    const comp = renderer.create(<Title size="xl" />).toJSON();

    expect(comp).toMatchSnapshot();
  });

  it('Renders with size lg', () => {
    const comp = renderer.create(<Title size="lg" />).toJSON();

    expect(comp).toMatchSnapshot();
  });

  it('Renders with children', () => {
    const comp = renderer.create(<Title>Hello</Title>).toJSON();

    expect(comp).toMatchSnapshot();
    expect(comp.children.length).not.toBe(0);
  });

  it('Renders with an unsupported size', () => {
    const comp = renderer.create(<Title size="md" />).toJSON();

    expect(comp).toMatchSnapshot();
  });
});
