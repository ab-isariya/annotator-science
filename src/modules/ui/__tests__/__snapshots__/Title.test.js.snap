// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title/> Renders with an unsupported size 1`] = `
<p
  className="font-inter font-light text-base leading-6 undefined"
/>
`;

exports[`<Title/> Renders with children 1`] = `
<h1
  className="font-inter font-light text-2xl leading-9 undefined"
>
  Hello
</h1>
`;

exports[`<Title/> Renders with size lg 1`] = `
<h2
  className="font-inter font-light text-base leading-6 undefined"
/>
`;

exports[`<Title/> Renders with size xl 1`] = `
<h1
  className="font-inter font-light text-2xl leading-9 undefined"
/>
`;

exports[`<Title/> Renders without crashing 1`] = `
<h1
  className="font-inter font-light text-2xl leading-9 undefined"
/>
`;
